kmehr/src/main/kotlin/org/taktik/icure/be/ehealth/logic/kmehr/smf/impl/v23g/SoftwareMedicalFileImport.kt

```diff
@@ -416,37 +416,42 @@ class SoftwareMedicalFileImport(
+        val relatedItemIds = kmehrIndex.serviceFor[mfId]
+                ?.mapNotNull { mf -> kmehrIndex.itemIds[mf]?.let { (mf to it) } }
+        if(!relatedItemIds.isNullOrEmpty()){
+            return relatedItemIds.map { (heOrHcaMfid, heOrHcaPair) ->
                 val item = heOrHcaPair.second
                 if (item.cds.find { it.s == CDITEMschemes.CD_ITEM }?.value == "healthcareapproach") {
                     val heId = kmehrIndex.approachFor[heOrHcaMfid]?.firstNotNullOfOrNull { kmehrIndex.itemIds[it] }?.first
                     SubContact(
-                        id = UUID.nameUUIDFromBytes(("$contactId|$heId|${heOrHcaPair.first}|null").toByteArray()).toString(),
-                        formId = formId,
-                        healthElementId = heId?.toString(),
-                        planOfActionId = heOrHcaPair.first.toString(),
-                        services = listOf(ServiceLink(serviceId = service.id)),
+                            id = UUID.nameUUIDFromBytes(("$contactId|$heId|${heOrHcaPair.first}|null").toByteArray()).toString(),
+                            formId = formId,
+                            healthElementId = heId?.toString(),
+                            planOfActionId = heOrHcaPair.first.toString(),
+                            services = listOf(ServiceLink(serviceId = service.id)),
                     )
                 } else {
                     SubContact(
-                        id = UUID.nameUUIDFromBytes(("$contactId|null|${heOrHcaPair.first}|null").toByteArray()).toString(),
-                        formId = formId,
-                        healthElementId = heOrHcaPair.first.toString(),
-                        services = listOf(ServiceLink(serviceId = service.id)),
+                            id = UUID.nameUUIDFromBytes(("$contactId|null|${heOrHcaPair.first}|null").toByteArray()).toString(),
+                            formId = formId,
+                            healthElementId = heOrHcaPair.first.toString(),
+                            services = listOf(ServiceLink(serviceId = service.id)),
                     )
                 }
-        } ?: formId?.let {
-            listOf(
-                SubContact(
-                    id = UUID.nameUUIDFromBytes(("$contactId|null|null|$formId").toByteArray()).toString(),
-                    formId = formId,
-                    services = listOf(ServiceLink(serviceId = service.id)),
-                ),
+            }
+        }
+        else if(!formId.isNullOrBlank()){
+            return listOf(
+                    SubContact(
+                            id = UUID.nameUUIDFromBytes(("$contactId|null|null|$formId").toByteArray()).toString(),
+                            formId = formId,
+                            services = listOf(ServiceLink(serviceId = service.id)),
+                    ),
             )
         }
+        return listOf()
+    }
```

Comment:
Could you change it to a `when` with a `return` outside the clauses? It will improve readability. I.e.:
```
return when {
    !relatedItemIds.isNullOrEmpty() -> relatedIds.map { /* the rest of the code */ }
    !formId.isNullOrBlank() -> listOf(/*the rest of the code */)
    else -> listOf()
}
```